### PR TITLE
chore: Update contrib doc and justfile commands.

### PR DESCRIPTION
**Pull request by milesj**
### Description

This PR does a few things:

- Updates our contributing doc with more standardized information.
  - What other info should we add?
- Updates our just commands to have rust/node specific variants.
- Updates CI to use just commands when applicable.
- Adds typescript checking.
- Fixes prettier checking.

### Test Plan

In CI.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -54,20 +39,21 @@ jobs:
       - if: steps.filter.outputs.src == 'true'
         run: cargo deny check
 
-  cargo-clippy:
-    name: Cargo Clippy
+  cargo-lint:
```
Comment (milesj) [inline .github/workflows/ci.yaml]: merged format/clippy

File: crates/rolldown/examples/basic.rs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-use rolldown::{Bundler, InputItem, InputOptions};
+use rolldown::{Bundler, InputItem, InputOptions, OutputOptions};
```
Comment (milesj) [inline crates/rolldown/examples/basic.rs:1]: these are fixing lint errors

File: packages/node/src/rolldown-build.ts [hunk 3]
```diff
@@ -3,23 +3,21 @@ import { normalizeOutputOptions, OutputOptions } from './options/output-options'
 import type { RollupBuild, SerializedTimings } from './rollup-types'
 import { transformToRollupOutput, RolldownOutput, unimplemented } from './utils'
 
-export class RolldownBuild implements RollupBuild {
+export class RolldownBuild implements Omit<RollupBuild, 'generate' | 'write'> {
```
Comment (milesj) [inline packages/node/src/rolldown-build.ts:6]: fixes some ts issues now that we run the typechecker

File: justfile [hunk 4]
```diff
@@ -41,10 +87,3 @@ setup-bench:
 
 bench:
     cargo bench -p bench
-
-# This command will try to run checks similar to ci locally
-check:
```
Comment (milesj) [inline justfile:46]: do we still need this?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+### Open development
+
+All development happens directly on GitHub. Both core team members and external contributors (via forks)
+send pull requests which go through the same review process.
+
+### Branch organization
+
+Submit all pull requests directly to the `main` branch. We only use separate branches for upcoming
+releases / breaking changes, otherwise, everything points to master.
```
Comment (Boshen) [inline CONTRIBUTING.md]: Points to main?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+Code that lands in main must be compatible with the latest stable release. It may contain
+additional features, but no breaking changes. We should be able to release a new minor version from
+the tip of main at any time.
+
+> Developers use (or may use) [Graphite](https://graphite.dev/) for better branch management.
```
Comment (Boshen) [inline CONTRIBUTING.md]: Unfortunately due to permission issues only people with write permission are able to use graphite.

Comment (milesj) [reply CONTRIBUTING.md]: TIL, fixed

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (Boshen) [reply .github/workflows/ci.yaml]: It'll require two pushes if the contributor didn't fmt nor clippy their code though 😅

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+## Workflow
+
+The following commands are available and should be used in your standard development workflow.
+
+- `just init` - Install required tools and dependencies.
+- `just check` - Runs the typechecker.
+- `just lint` - Lints code.
+- `just fmt` - Formats code.
+- `just test` - Runs tests.
```
Comment (Boshen) [inline CONTRIBUTING.md:72]: I just noticed that we don't have a `just ready`, for which I love!

In oxc:

```
# When ready, run the same CI commands
ready:
  git diff --exit-code --quiet
  typos
  cargo fmt
  just check
  just test
  just lint
  git status
```

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (milesj) [reply .github/workflows/ci.yaml]: true... didnt think about that. i guess the same is true for node too.

File: CONTRIBUTING.md (hunk 7 above)
Comment (milesj) [reply CONTRIBUTING.md:72]: Added!

File: package.json [hunk 8]
```diff
@@ -24,11 +24,12 @@
...
     "test:update-rollup": "yarn workspace rollup-tests run test:update",
-    "prettier": "prettier . '!**/*.{js||ts|json|md|yml|yaml}' -w",
-    "prettier:ci": "prettier . '!**/*.{js||ts|json|md|yml|yaml}' -c",
+    "prettier": "prettier . '!**/*.{js,ts,json,md,yml,yaml}' -w",
+    "prettier:ci": "prettier . '!**/*.{js,ts,json,md,yml,yaml}' -c",
+    "typecheck": "yarn workspace @rolldown/node run typecheck",
     "bench": "yarn workspace bench run bench",
     "TODO(hyf0): #need to investigate following commands": "_",
-    "build:ci:release": "run-s build:packages:ci build:ts build:binding:release",
+    "build:ci:release": "yarn workspaces foreach -R --topological-dev --from '@rolldown/node' run build",
```
Comment (underfin) [inline package.json]: Here need using `build:release`.

File: .github/workflows/ci.yaml [hunk 9]
```diff
@@ -83,26 +84,21 @@ jobs:
...
 
   node-test:
     name: Node Test
     runs-on: buildjet-2vcpu-ubuntu-2204
     timeout-minutes: 30
     strategy:
       fail-fast: false
-      matrix:
-        os:
-          - ubuntu-latest
```
Comment (underfin) [inline .github/workflows/ci.yaml:102]: Could you preserve it at now, because we need to enable it at publish repo.

File: package.json (hunk 8 above)
Comment (milesj) [reply package.json]: @underfin I didn't change this, I think some script is automatically injecting it.

**Review by underfin — APPROVED** [review]

**Pull request merged**